### PR TITLE
[WIP] Link

### PR DESCRIPTION
**Pull request by ghost**
This is kinda a mess. Anyway it seems to work when I call "findIndexNode" after writing the new inode, otherwise it forgets that it was assigned an inode and remains -1. I'll try to figure out what's happening later

I can understand if you don't want to read through all the code because it's really long, but if you have any ideas let me know.

### CODE REVIEW

File: Kernel.cc [hunk 1]
```diff
@@ -1367,3 +1370,277 @@ short Kernel::findIndexNode( char * path , IndexNode& inode )
 	return indexNodeNumber ;
 }
 
+int Kernel::link(char *oldpath, char *newpath) {
+  	// get the full path
+	char *fullPath = getFullPath(oldpath);
```
Comment (ghost) [inline Kernel.cc:1375]: This works exactly like create, by traversing the directory. There's probably a better way to do it

File: Kernel.cc [hunk 2]
```diff
@@ -1367,3 +1370,277 @@ short Kernel::findIndexNode( char * path , IndexNode& inode )
...
+		    // if no entry read, write the new item at the current 
+		    // location and break
+		    writedir( dir , newDirectoryEntry) ;
+		    IndexNode note;
+		    
+		    static char funPath[1024];
+		    memset(fullPath, '\0', 1024);
+		    strcpy(funPath, "/away/jon/new.txt");
+		    short num = findIndexNode(funPath, note);
+		    //cout << "num is: " << num << endl;
```
Comment (ghost) [inline Kernel.cc:1556]: This is the line that is needed for some reason. It shouldn't be needed, but it's the only the filesystem seems to remember the inode number

Comment (elliott-beach) [reply Kernel.cc:1556]: What is `/away/jon/new.txt`? It's not obvious to me what most of this is doing. I'm also not sure which line you're referring to, and why it shouldn't be needed.

File: Kernel.cc [hunk 3]
```diff
@@ -1367,3 +1370,277 @@ short Kernel::findIndexNode( char * path , IndexNode& inode )
...
+
+		// current item = next item
+		nextDirectoryEntry.copy(currentDirectoryEntry);
+	    } 
+
+	    // close the directory
+	    close(dir) ;
+	}
+	// Assume it doesn't exist in the filesystem yet
+	/*else
```
Comment (elliott-beach) [inline Kernel.cc:1602]: we should return -1 in this case.

File: Kernel.cc (hunk 2 above)
Comment (elliott-beach) [reply Kernel.cc:1556]: Closed??? Seems harsh bro

**Pull request closed without merging**